airbyte-integrations/connectors/source-trello/source_trello/components.py

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from dataclasses import dataclass
+from typing import Iterable
+
+import dpath.util
+from airbyte_cdk.models import AirbyteMessage, SyncMode, Type
+from airbyte_cdk.sources.declarative.partition_routers.substream_partition_router import SubstreamPartitionRouter
+from airbyte_cdk.sources.declarative.types import Record, StreamSlice
+
+
+@dataclass
+class OrdersIdPartitionRouter(SubstreamPartitionRouter):
+
+    def stream_slices(self) -> Iterable[StreamSlice]:
+        """
+        Iterate over each parent stream's record and create a StreamSlice for each record.
+
+        For each stream, iterate over its stream_slices.
+        For each stream slice, iterate over each record.
+        yield a stream slice for each such records.
+
+        If a parent slice contains no record, emit a slice with parent_record=None.
+
+        The template string can interpolate the following values:
+        - parent_stream_slice: mapping representing the parent's stream slice
+        - parent_record: mapping representing the parent record
+        - parent_stream_name: string representing the parent stream name
+        """
+        yielded_board_ids = set() # to track the board ids
+        config_board_ids = set(self.config.get("board_ids", [])) # board ids specified in config
+        for parent_stream_config in self.parent_stream_configs:
```

Comment:
This function is huge! Is there anything we can do maybe to break it? Trying to think the real need of it :(

Reply:
Please review now @marcosmarxm. I have simplified the function.